### PR TITLE
Feature/wcms 223 queue operations

### PR DESCRIPTION
**Pull request by pingers**
This PR affects backup and restore operations only.

Triggering a backup/restore adds the operation to the queue. 
```select * from queue;```

In order to perform the actual operation, you must run cron.
Operations block per environment and the active jobs can be inspected using:
```select * from key_value where name like 'shp_job.*';```

Clearing the active job is a bit weird atm and won't occur until the next job for that environment attempts to run. I'm sure we can do better. Ideas?

Test coverage could be improved, but at least covers the active job api. From `web/core`:
```../../bin/phpunit --group=shepherd```

### CODE REVIEW

File: web/modules/custom/shp_backup/shp_backup.services.yml [hunk 1]
```diff
@@ -1,13 +1,7 @@
 services:
   shp_backup.backup:
     class: Drupal\shp_backup\Service\Backup
-    arguments: ['@config.factory', '@token', '@entity_type.manager']
-  shp_backup.environment_restore_form:
```
Comment (pingers) [inline web/modules/custom/shp_backup/shp_backup.services.yml]: None of the forms are services... bye!

Comment (bart88) [reply web/modules/custom/shp_backup/shp_backup.services.yml]: 👍

File: web/modules/custom/shp_backup/src/Service/Backup.php [hunk 2]
```diff
@@ -91,18 +124,21 @@ public function getAll(NodeInterface $site) {
...
    *
-   * @return int|bool
-   *   SAVED_NEW or FALSE.
+   * @return \Drupal\Core\Entity\EntityInterface|static
+   *   The backup entity.
    */
   public function createNode(NodeInterface $environment, $title = NULL) {
     if (!isset($title)) {
+
+      // @todo Inject the service.
```
Comment (bart88) [inline web/modules/custom/shp_backup/src/Service/Backup.php]: Yes please.

File: web/modules/custom/shp_backup/src/Service/QueuedBackup.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+   * @param \Drupal\node\NodeInterface $backup
+   *   The environment to backup.
+   *
+   * @return bool
+   *   True if queued.
+   */
+  public function create(NodeInterface $backup) {
+    $node_storage = $this->entityTypeManager->getStorage('node');
+    $environment = $node_storage->load($backup->field_shp_environment->target_id);
+    return \Drupal::service('shp_orchestration.job_queue')->add(
```
Comment (bart88) [inline web/modules/custom/shp_backup/src/Service/QueuedBackup.php]: Use the jobQueue service you have already injected.

File: web/modules/custom/shp_backup/src/Service/QueuedBackup.php [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+   * @param \Drupal\node\NodeInterface $environment
+   *   The environment to restore.
+   *
+   * @return bool
+   *   True if queued.
+   */
+  public function restore(NodeInterface $backup, NodeInterface $environment) {
+    $node_storage = $this->entityTypeManager->getStorage('node');
+    $source_environment = $node_storage->load($backup->field_shp_environment->target_id);
+    return \Drupal::service('shp_orchestration.job_queue')->add(
```
Comment (bart88) [inline web/modules/custom/shp_backup/src/Service/QueuedBackup.php]: Use the jobQueue service you have already injected.

File: web/modules/custom/shp_backup/src/ShpBackupServiceProvider.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public function alter(ContainerBuilder $container) {
+    if ($this->isQueueEnabled()) {
+      // Overrides backup service to use a queued backup service.
+      $definition = $container->getDefinition('shp_backup.backup');
+      $definition->setClass('Drupal\shp_backup\Service\QueuedBackup');
+      $arguments   = $definition->getArguments();
```
Comment (bart88) [inline web/modules/custom/shp_backup/src/ShpBackupServiceProvider.php]: Tooo much space.

File: web/modules/custom/shp_orchestration/tests/src/Unit/ActiveJobManagerTest.php [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+/**
+ * Tests the HeadingIdFilter class.
+ *
+ * @group ua
+ * @group shepherd
+ * @group shp_orchestration
+ * @coversDefaultClass \Drupal\shp_orchestration\Service\ActiveJobManager
+ */
+class ActiveJobManagerTest extends UnitTestCase {
```
Comment (bart88) [inline web/modules/custom/shp_orchestration/tests/src/Unit/ActiveJobManagerTest.php]: 👍 💯  For providing a unit test

**Comment by singularo** [comment]
Awesome, works a treat!
:ship:

File: web/modules/custom/shp_backup/src/Service/QueuedBackup.php (hunk 3 above)
Comment (pingers) [reply web/modules/custom/shp_backup/src/Service/QueuedBackup.php]: TY! This was pretty dumb :small_airplane:

**Pull request merged**